### ISSUE TITLE
Brushes transparent color should be the one from grab time, not the current backcolor.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [birdal46/grafx2#81]
```
Grafx2 uses the current background color to determine which part of a brush
is transparent.
In comparison, Deluxe Paint III - IV used the color index that was the
background color at the time the brush was grabbed: what was
transparent/solid when you grabbed the brush stay this way no matter your
future pen colors.
Still years after moving from DP to grafx2, I often find cases where I wish
grafx2 behaved the DP II way about this.

This is especially true for monochrome custom brushes: Once you have your
shape in brush, you can freely change the foreground color and paint with
it, but you can't freely paint with the right mouse button, as any change
of background color (for future painting) will additionally change the
brush shape.

1) For monochromized brushes, it seems always better : this way they behave
more like the resizable monochrome brushes.
2) For color brushes, I'm not sure... the current system has its advantage,
it allows making any brush "solid" by choosing a BG color that isn't in the
brush. and transparent again by re-selecting a color.
3b) But then, in DP mode, on a single keyboard shortcut it's trivial to
program a function that picks an unused color. And to complete the
functionality, a tool like the color picker to pick the new transparent color.

In short, I'm not decided but I'll definitely experiment with the DP method.

```

Original issue reported on code.google.com by `yrizoud` on 29 Dec 2008 at 2:53

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
Original comment by `pulkoma...@gmail.com` on 26 May 2009 at 7:50
- Added labels: **Component-Pages**, **Milestone-2.1**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
Original comment by `pulkoma...@gmail.com` on 15 Sep 2009 at 7:13
- Added labels: **Milestone-2.3**

**Comment by GoogleCodeExporter** [comment]
```
Actually I was bothered with this behavior initially, but now I've grown used 
to it.
Should you decide to alter this, please make it a toggle.
```

Original comment by `ilija.melentijevic` on 15 Nov 2009 at 10:36

**Comment by GoogleCodeExporter** [comment]
```
I think it could be nice to have an extra functionality :
 * Keep everything working as usual
 * Add a way to select a new transparent color, and replace the old one with the new 
one in the brush. This way, you can make multiple colors transparent to remove 
parts 
of the brush in sequence.
```

Original comment by `pulkoma...@gmail.com` on 5 Apr 2010 at 10:18

**Comment by GoogleCodeExporter** [comment]
```
What you describe can be done manually by using the spare page: You paste the 
brush,
perform a few replacements, and grab it again. This method gives you the right 
visual
tools (Fill, Color Replacer, any drawing tools) and you can Undo/Redo as many 
steps
as you want.
```

Original comment by `yrizoud` on 5 Apr 2010 at 11:57

**Comment by GoogleCodeExporter** [comment]
```
Yes.
Then i'd like something else : OCP art studio on amstrad has a "last window" 
tool 
that will re-grab the brush from where it was picked. This allows for pasting a 
brush, altering it, then grabbing it again quickly. As grabbing a brush at the 
exact 
same pixels isn't always that easy...
```

Original comment by `pulkoma...@gmail.com` on 5 Apr 2010 at 12:14

**Comment by GoogleCodeExporter** [comment]
Original comment by `pulkoma...@gmail.com` on 20 Jan 2011 at 8:42
- Changed title: **Brushes transparent color should be the one from grab time, not the current backcolor.**

**Comment by GoogleCodeExporter** [comment]
Original comment by `pulkoma...@gmail.com` on 15 Feb 2011 at 8:12
- Added labels: **Milestone-2.4**

**Comment by GoogleCodeExporter** [comment]
```
Arbitering issues that make it to v2.4
```

Original comment by `yrizoud` on 8 Mar 2012 at 7:18
- Added labels: **Milestone-2.5**
